lib/airbrake/rack/user.rb

```diff
@@ -1,30 +1,33 @@
 module Airbrake
   module Rack
     ##
-    # Represents an authenticated Warden user, which can be converted to
-    # Airbrake's payload format.
+    # Represents an authenticated user, which can be converted to Airbrake's
+    # payload format. Supports Warden and Omniauth authentication frameworks.
     class User
-      # Finds the Warden user in the Rack environment and creates a new user
-      # wrapper.
+      # Finds the user in the Rack environment and creates a new user wrapper.
       #
       # @param [Hash{String=>Object}] rack_env The Rack environment
       # @return [Airbrake::Rack::User, nil]
       def self.extract(rack_env)
-        return unless (warden = rack_env['warden'])
-        return unless (warden_user = warden.user(run_callbacks: false))
-        new(warden_user)
+        # Warden support (including Devise).
+        if (warden = rack_env['warden'])
+          if (user = warden.user(run_callbacks: false))
+            return new(user) if user
+          end
+        end
+
+        # Fallback mode (OmniAuth support included). Works only for Rails.
+        controller = rack_env['action_controller.instance']
```

Comment:
Shouldn't this be env instead of rack_env if not using Warden? I have a Rails app with current_user in my controller but airbrake is unable to locate it because it tries to fetch rack_env instead of env.